File: backend/internal/core/domain/k8s.go
package domain

import "time"

// ─── Platform hub (kubernetes server) responses ───────────────────────────────

type K8sGateway struct {
	Name       string `json:"name"`
	Namespace  string `json:"namespace"`
	Address    string `json:"address"`
	Programmed bool   `json:"programmed"`
}

type K8sRoute struct {
	Name       string     `json:"name"`
	Namespace  string     `json:"namespace"`
	Gateway    string     `json:"gateway"`
	Hostnames  []string   `json:"hostnames"`
	Links      []string   `json:"links"`               // https://<host> per hostname
	CertReady  *bool      `json:"certReady,omitempty"` // nil = no matching cert found
	CertExpiry *time.Time `json:"certExpiry,omitempty"`
}

// K8sRoutesResponse powers the "directory" tiles + gateway status.
type K8sRoutesResponse struct {
	Gateways []K8sGateway `json:"gateways"`
	Routes   []K8sRoute   `json:"routes"`
}

type K8sNode struct {
	Name           string `json:"name"`
	KubeletVersion string `json:"kubeletVersion"`
	Ready          bool   `json:"ready"`
}

type K8sWorkload struct {
	Name      string `json:"name"`
	Namespace string `json:"namespace"`
	Ready     int    `json:"ready"`
	Desired   int    `json:"desired"`
	Healthy   bool   `json:"healthy"`
}

type K8sCert struct {
	Name      string     `json:"name"`
	Namespace string     `json:"namespace"`
	DNSNames  []string   `json:"dnsNames"`
	NotAfter  *time.Time `json:"notAfter,omitempty"`
	DaysLeft  *int       `json:"daysLeft,omitempty"`
	Ready     bool       `json:"ready"`
}

type K8sDatastore struct {
	Kind      string `json:"kind"` // "postgres-cnpg"
	Name      string `json:"name"`
	Namespace string `json:"namespace"`
	Instances int    `json:"instances"`
	Ready     int    `json:"ready"`
	Phase     string `json:"phase"`
	Healthy   bool   `json:"healthy"`
}

// K8sHealth is the cluster health strip: nodes, workloads, certs, datastores.
type K8sHealth struct {
	Nodes      []K8sNode      `json:"nodes"`
	Workloads  []K8sWorkload  `json:"workloads"`
	Certs      []K8sCert      `json:"certs"`
	Datastores []K8sDatastore `json:"datastores"`
}

File: app/src/types/k8s.ts
export interface K8sGateway {
  name: string;
  namespace: string;
  address: string;
  programmed: boolean;
}

export interface K8sRoute {
  name: string;
  namespace: string;
  gateway: string;
  hostnames: string[];
  links: string[];
  certReady?: boolean | null;
  certExpiry?: string | null;
}

export interface K8sRoutesResponse {
  gateways: K8sGateway[];
  routes: K8sRoute[];
}

export interface K8sNode {
  name: string;
  kubeletVersion: string;
  ready: boolean;
}

export interface K8sWorkload {
  name: string;
  namespace: string;
  ready: number;
  desired: number;
  healthy: boolean;
}

export interface K8sCert {
  name: string;
  namespace: string;
  dnsNames: string[];
  notAfter?: string | null;
  daysLeft?: number | null;
  ready: boolean;
}

export interface K8sDatastore {
  kind: string;
  name: string;
  namespace: string;
  instances: number;
  ready: number;
  phase: string;
  healthy: boolean;
}

export interface K8sHealth {
  nodes: K8sNode[];
  workloads: K8sWorkload[];
  certs: K8sCert[];
  datastores: K8sDatastore[];
}

export type IntegrationAuthMethod = "none" | "basic" | "bearer" | "header";

export interface Integration {
  id: string;
  serverId: string;
  kind: string;
  name: string;
  url: string;
  authMethod: IntegrationAuthMethod;
  hasSecret: boolean;
  hidden: boolean;
  createdAt: string;
}

export interface CreateIntegrationPayload {
  kind: string;
  name: string;
  url: string;
  authMethod?: IntegrationAuthMethod;
  secret?: string;
}

export interface UpdateIntegrationPayload {
  name: string;
  url: string;
  authMethod?: IntegrationAuthMethod;
  hidden?: boolean;
  secret?: string;
}
